src/main/java/duke/tasks/TaskList.java

```diff
@@ -0,0 +1,276 @@
+                );
+            } catch (EmptyListException e) {
+            }
+        } else if (command.equals(DONE)) {
+            try {
+                markAsDone(userInput);
+            } catch (DoneFormatException e) {
+                System.out.println("OOPS!!! Done command must follow the format: done <index>");
+            } catch (DoneAlreadyException e) {
+                System.out.println("OOPS!!! The task has already been completed!");
+            } catch (DoneRangeException e) {
+                System.out.println("OOPS!!! The index is out of range for the number of tasks! Please enter a valid index");
+            }
+        } else if (command.equals(TODO)) {
+            try {
+
+                addTodo(userInput);
+            } catch (TodoException e) {
+                System.out.println("OOPS!!! duke.tasks.Todo command must follow the format: todo <task description>");
+            }
+        } else if (command.equals(EVENT)) {
+            try {
+                String taskDescription = userInput.replaceAll(GET_DESCRIPTION_DELIMITER, "");
+                String at = userInput.replaceAll(GET_AT_DELIMITER, "");
+                addEvent(taskDescription, at);
+            } catch (EventException e) {
+                System.out.println("OOPS!!! duke.tasks.Event command must follow the format: <task description> /at <time/date>");
+            }
+        } else if (command.equals(DEADLINE)) {
+            try {
+                String taskDescription = userInput.replaceAll(GET_DESCRIPTION_DELIMITER, "");
+                String by = userInput.replaceAll(GET_BY_DELIMITER, "");
+                addDeadline(taskDescription, by);
+            } catch (DeadlineException e) {
+                System.out.println("OOPS!!! duke.tasks.Deadline command must follow the format: <task description> /by <time/date>");
+            }
+        }
+    }
+
+
+    public static void printAcknowledgement() {
+        int taskCount = taskList.size();
+        String latestTaskAdded = taskList.get(taskCount - 1).showTask();
+        System.out.println("Got it. I've added this task:\n  " +
+                latestTaskAdded +
+                "\nNow you have " + taskCount + " tasks in the list"
+        );
+    }
+
+    public static String getAcknowledgement() {
+        int taskCount = taskList.size();
+        String latestTaskAdded = taskList.get(taskCount - 1).showTask();
+        return "Got it. I've added this task:\n  " +
+                latestTaskAdded +
+                "\nNow you have " + taskCount + " tasks in the list";
+    }
+
+    /**
+     * Add todo task
+     */
+    public static void addTodo(String taskDescription) throws TodoException {
+
+        Task newTask = new Todo(taskDescription);
+        taskList.add(newTask);
+        System.out.println(DIVIDER);
+        printAcknowledgement();
+        System.out.println(DIVIDER);
+    }
+
+    public static String addTodo2(String commandArgument) throws TodoException {
```

Comment:
Not sure why you have 2 `addTodo` functions? If both the functions are performing different operations then please name them according to what they do. (Similar for your `addEvent`, `addDeadline` and `markAsDone` functions)